iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -2098,9 +2098,10 @@ public boolean canPerformMetadataDelete(org.apache.hadoop.hive.ql.metadata.Table
   }
 
   @Override
-  public List<FieldSchema> getPartitionKeys(org.apache.hadoop.hive.ql.metadata.Table hmsTable) {
+  public List<FieldSchema> getPartitionKeys(org.apache.hadoop.hive.ql.metadata.Table hmsTable, boolean latestSpecOnly) {
     Table icebergTable = IcebergTableUtil.getTable(conf, hmsTable.getTTable());
-    return IcebergTableUtil.getPartitionKeys(icebergTable, icebergTable.spec().specId());
+    return latestSpecOnly ? IcebergTableUtil.getPartitionKeys(icebergTable, true) :
```

Comment:
why do you need ternary operator here?
````IcebergTableUtil.getPartitionKeys(icebergTable, latestSpecOnly)````

Reply:
No need. :) Fixed.